Complete HW2

pisohaka opened the pull request:
# Выполнено ДЗ №

 - [X] Основное ДЗ
 - [X] Задание со *

## В процессе сделано:
1. Create frontend Deployment and ReplicaSet manifests
2. Build paymentservice docker container, push it on docker.io
3. Create paymentservice Deployment manifest, create manifests for blue-green and reverse deployments
4. Add readinessProbe to frontend-deployment.yaml
5. Create node-exporter DaemonSet manifest and edited it to monitor master nodes

## Как запустить проект:
 * Запустить команды:
git clone https://github.com/otus-kuber-2020-04/pisohaka_platform
cd pisohaka_platform
git checkout kubernetes-controllers

## Как проверить работоспособность:
kubectl get pods

## PR checklist:
 - [x] Выставлен label с темой домашнего задания

pisohaka committed:
Complete HW2

pisohaka committed:
Fix kubernetes-controllers dir name

Jasstkn commented on kubernetes-controllers/node-exporter-daemonset.yaml:
```diff
@@ -0,0 +1,63 @@
+      tolerations:
+      # this toleration is to have the daemonset runnable on master nodes
+      # remove it if your masters can't run pods
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
```
Вы правильно определили tolerations для данного случая, но данное решение не является универсальным.
Предлагаю вам попробовать сделать его таковым, а для этого подробнее ознакомиться с документацией:
https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/#concepts -> секция Note и с официальным манифестом для node exporter.

pisohaka replied:
Hi @Jasstkn, I changed tolerations options to operator: Exists with no key.  An empty key with operator Exists matches all keys, values and effects which means this will tolerate everything. So I think that the problem was solved, and I made manifest more universal

Jasstkn replied:
Exactly what we need. Good work, approved!

Jasstkn reviewed (changes requested):
Добрый день.
Домашнее задание выполнено верно - предлагаю вам попробовать найти универсальное решение для запуска node-exporter на всех нодах (подробности в комментарии к соответствующему файлу). По остальным заданиям все верно - отличная работа!

pisohaka committed:
Change tolerations option for node-exporter manifest to Exists

Change tolerations option for node-exporter manifest to Exists because
Exists matches all keys, values and effects which means this will tolerate everything.
